extensions/wikia/GlobalWatchlist/GlobalWatchlistBot.class.php

```diff
@@ -0,0 +1,432 @@
+<?php
+
+use \Wikia\Tasks\AsyncTaskList;
+use \Wikia\Logger\WikiaLogger;
+
+class GlobalWatchlistBot {
+
+	const MAX_ARTICLES_PER_WIKI = 50;
+
+	public function __construct() {
+		global $wgExtensionMessagesFiles;
+		$wgExtensionMessagesFiles['GlobalWatchlist'] = dirname( __FILE__ ) . '/GlobalWatchlist.i18n.php';
+	}
+
+	/**
+	 * send email to user
+	 */
+	public function sendDigestToUser( $userID ) {
```

Comment:
I wouldn't say "easily." This function and the `composeMail` one below it are pretty hairy. I ripped out about 60% of this file already to clean it up, and rebuilt the system which ultimately calls this, but don't think I have it in me to wade through these methods right now. How about I add a couple of TODO's to try and inspire the next person who touches this file to break those up a bit?